packages/synapse-react-client/src/mocks/msw/handlers/tableQueryHandlers.ts

```diff
@@ -54,14 +54,15 @@ export function getHandlersForTableQuery(
   response: QueryResultBundle,
   backendOrigin = getEndpoint(BackendDestinationEnum.REPO_ENDPOINT),
 ) {
-  const asyncJobId = uniqueId()
-  let queryBundleRequest: QueryBundleRequest | undefined
+  // We must map generated async job IDs to requests to disambiguate multiple calls to these endpoints
+  const mapOfRequests = new Map<string, QueryBundleRequest>()
```

Comment:
If we have multiple table query requests, keep track of IDs so the correct response is returned.